app/scripts/lib/segment.js

```diff
@@ -99,6 +101,12 @@ export function getTrackSegmentEvent (
       trackOptions.userId = metaMetricsId
     }
 
+    if (referrerUrl) {
+      trackOptions.referrer = {
```

Comment:
This should be in the `context` property, not in the top-level track options

```suggestion
      trackOptions.context.referrer = {
```